metabase/metabase
# FE: Update data model page for json fields

## Changes

[Product Doc](https://www.notion.so/metabase/Add-support-to-records-in-JSON-fields-8333f4db842a4bcfa449a09dbdcdab0c)
[Mockups](https://www.figma.com/file/UPmKgv5KW7L9M9jC0ygnj0?embed_host=notion&kind=&node-id=317%3A6517)

Update the Data Model page to support JSON fields. E2E specs will be added later due to the need to extend the test dataset.

<img width="1331" alt="Screenshot 2022-04-13 at 23 59 47" src="https://user-images.githubusercontent.com/14301985/163260585-46b65175-0bcf-43f3-b7bf-41757c0aea81.png">

<img width="1319" alt="Screenshot 2022-04-13 at 23 59 59" src="https://user-images.githubusercontent.com/14301985/163260543-de22fd34-6cc5-46d3-989e-2525e86514d4.png">

## How to verify

- Add Postgres DB with a json fields
- Open the data model page of the table containing JSON fields
- Ensure it shows raw name on top of each column card 
- Select "Original schema" tab and ensure it shows nested fields in a grouped way
